components/filter/filter.js

```diff
@@ -607,7 +613,26 @@ class Filter extends FocusMixin(LocalizeCoreElement(RtlMixin(LitElement))) {
 	}
 
 	_getActiveDimension() {
-		return !this._activeDimensionKey ? this._dimensions[0] : this._dimensions.find(dimension => dimension.key === this._activeDimensionKey);
+		return !this._activeDimensionKey ? this._dimensions[0] : this._getDimensionByKey(this._activeDimensionKey);
+	}
+
+	_getDimensionByKey(key) {
+		return this._dimensions.find(dimension => dimension.key === key);
+	}
+
+	_getSearchCallback(dimension) {
+		return function({ keysToDisplay = [], displayAllKeys = false } = {}) {
+			requestAnimationFrame(() => {
+				dimension.displayAllKeys = displayAllKeys;
+				dimension.searchKeysToDisplay = keysToDisplay;
+				this._performDimensionSearch(dimension);
+				dimension.loading = false;
+				if (dimension.selectedFirst) {
+					this._updateDimensionShouldBubble(dimension);
+				}
```

Comment:
Are we now running this twice for the search scenario (once [here](https://github.com/BrightspaceUI/core/blob/518f5c4264aa3825ff02e06f35c438579f572a7c/components/filter/filter.js#L800-L802), and then again above)?

If so, could we move it into `_performDimensionSearch`? Seems like that hits all paths - automatic search, manual search, and search after load more?